sequencer.go

```diff
@@ -18,18 +26,36 @@ func NewSeedSequencer(idx Index, src ...Seed) *SeedSequencer {
 	}
 }
 
+// Plan returns an array of SeedSegmentCandidate, representing an ordered list of
+// segments that can be used to re-assemble the requested file
+func (r *SeedSequencer) Plan() (plan []SeedSegmentCandidate) {
+	for {
+		seed, segment, source, done := r.Next()
+		plan = append(plan, SeedSegmentCandidate{seed, source, segment})
```

Comment:
If you look at the previous code, I believe the append should happen after "done" is checked. Looks like when done == true then there is no valid seed in the response (best to verify that). Should be able to make a test where the seed matches the target file and basically the whole file is restored from one segment first, then in the 2nd step "done" should be true but no more seed to use.

Reply:
In `assemble.go` there was:
```
		chunks, from, done := seq.Next()
		select {
		case <-ctx.Done():
			break loop
		case in <- Job{chunks, from}:
		}
		if done {
			break
		}
```
And `Next()` returns `segment, source, r.current >= len(r.index.Chunks)`, so the last valid segment is expected to have `done` set to true.

With this info I guess that we need to append and then check the `done`  value.
Does it make sense?

Reply:
Oh, yes, you're correct. I got that mixed up with the .Done() from the context. So first append, then check `done` is right